packages/onboarding/src/scroll-to-top/index.tsx

```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+import { useLocation } from 'react-router-dom';
+
+type ScrollToTopProps = {
+	selector: string;
+};
+
+const ScrollToTop: React.FunctionComponent< ScrollToTopProps > = ( { selector } ) => {
+	const { pathname } = useLocation();
+
+	React.useEffect( () => document.querySelector( selector )?.scrollTo( 0, 0 ), [
+		selector,
+		pathname,
+	] );
+
+	return null;
```

Comment:
Why aren't we using a custom hook instead of having a component returning `null`?

Reply:
Good point/spot @razvanpapadopol. @StefanNieuwenhuis could we update this to become a hook?

Reply:
We're unable to use a custom hook in the `<Router/>` and in order to comply with best practices, we have to use a component (see [react-router-dom docs](https://reactrouter.com/web/guides/scroll-restoration/scroll-to-top)). 

In order use-cases a hook absolutely makes sense but that's out of scope for this issue.